CONTRIBUTING.md

```diff
@@ -0,0 +1,110 @@
+# Rodio Development Guide
+
+## Quick Start
+
+1. Install [Rust](https://www.rust-lang.org/tools/install)
+2. Clone the repository: `git clone https://github.com/RustAudio/rodio.git`
+3. Navigate to the project: `cd rodio`
+4. Build the project: `cargo build`
+
+## Project Structure
+
+src/:
+- `source/`: Audio source implementations
+- `decoder/`: Audio format decoders
+- `sink/`: Audio playback and mixing
+- `dynamic_mixer/`: Real-time audio mixing
+- `spatial/`: Spatial audio capabilities
+
+## Coding Guidelines
+
+- Follow [Rust API Guidelines](https://rust-lang.github.io/api-guidelines/)
+- Use `rustfmt` for formatting
+- Implement `Source` trait for new audio sources
+- Use `Sink` for playback management
+
+## Common Tasks
+
+### Adding a New Audio Source or Effect
+
+1. Create a new file in `src/source/`
+2. Implement the `Source` trait to define how audio samples are generated or modified
+3. Consider implementing sources like oscillators, noise generators, or effects like amplification, filtering, or distortion
+4. Optimize for performance, especially for real-time processing in effects
+5. Add unit tests/benchmarks where applicable to ensure proper functionality and performance
+
+### Implementing a New Decoder
+
+1. Add new module in `src/decoder/`
+2. Implement necessary traits (e.g., `Decoder`) to handle specific audio formats
+3. Focus on efficiently parsing audio file headers and decoding compressed audio data
+4. Update `src/decoder/mod.rs` to integrate the new decoder
+
+## Testing
+
+- Write unit tests for each new function (if applicable)
```

Comment:
I actually dislike unit tests :sweat_smile:. They make it harder to refactor code since all the unit tests start breaking if you do. Given Rusts robust type system I also do not think they add a lot unlike in langs like python. 

Unit tests can be very valuable during development though! I just remove them after.

Reply:
Then ill mention that they can add them during development if it helps them, but must be removed before the final merge?

Reply:
yeah lets do that. "must" sounds so strict though, I want to communicate that they should not spend too much time on the unit tests, they may be very rough since they are temporary.

Reply:
Check the new changes and let me know if it seems OK